pkg/apis/relay.sh/v1beta1/tenant_types.go

```diff
@@ -56,7 +56,7 @@ type ToolInjection struct {
 	// populated and attached to every tenant container.
 	//
 	// +optional
-	VolumeClaimTemplate *corev1.PersistentVolumeClaim `json:"volumeClaimTemplate,omitempty"`
+	VolumeClaimTemplate *corev1.PersistentVolumeClaimSpec `json:"volumeClaimTemplate,omitempty"`
```

Comment:
I think we probably should leave this as the full object so we can propagate labels and annotations if a user wants to add them.

Reply:
I'll leave it for now then. It's certainly convenient. And it is consistent with a few examples I've seen (Tekton, StatefulSets, etc.)

Fundamentally, it's more of a template model, though, which means we should consider having a template object explicitly defined that allows `metadata` and `spec`. Including the `status` field seems a bit off, even if it is ignored and doesn't cause any issues.

Reply:
It's not super uncommon to see this elsewhere in Kubernetes (e.g., `volumeClaimTemplates` in https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#statefulsetspec-v1-apps) but yeah, it might be nice to explicitly set out what we use and don't use someday. I think it's fine for now though - users should find it to be pretty intuitive.

Reply:
StatefulSets are specifically interesting, because they use a PodTemplate (which is just the ObjectMeta and Pod), but for the same concept with the aptly named `volumeClaimTemplates` it references the entire PersistentVolumeClaim instead.

Reply:
It's probably clear enough that explicitly setting `status` won't get you anywhere...